aten/src/ATen/AccumulateType.h

```diff
@@ -25,15 +25,19 @@ template <> struct AccumulateType<half, true> { using type = float; };
 #endif
 template <> struct AccumulateType<Half, true> { using type = float; };
 template <> struct AccumulateType<float, true> { using type = float; };
+template <> struct AccumulateType<std::complex<float>, true> { using type = std::complex<float>; };
 template <> struct AccumulateType<double, true> { using type = double; };
+template <> struct AccumulateType<std::complex<double>, true> { using type = std::complex<double>; };
 template <> struct AccumulateType<int8_t, true> { using type = int64_t; };
 template <> struct AccumulateType<uint8_t, true> { using type = int64_t; };
 template <> struct AccumulateType<char, true> { using type = int64_t; };
 template <> struct AccumulateType<int16_t, true> { using type = int64_t; };
 template <> struct AccumulateType<int32_t, true> { using type = int64_t; };
 template <> struct AccumulateType<int64_t, true> { using type = int64_t; };
 template <> struct AccumulateType<BFloat16, false> { using type = float; };
+template <> struct AccumulateType<std::complex<float>, false> { using type = std::complex<float>; };
```

Comment:
why this choice?  It doesn't match with the non-complex case, i.e. float->double?

Reply:
which seems wrong if you consider passing complex values that have no imaginary component.

Reply:
@gchanan, thanks for your comments, we have also changed non-complex case for float, from double to float (float -> float), align CPU and CUDA design. so from your side, is there any reason that the accumulate type of float should be double?

@EikanWang  please help merge that changes into this PR. align complex design with non-complex one

Reply:
@hongzhen1:

1) You didn't change the non-complex case for float (see the line directly below this), so I'm not sure what this is referring to
2) You shouldn't change that in any case, that's a major design change that requires discussion -- if your goal is to improve accumulate kernels for complex, this patch should focus on that.